### PR TITLE
feat: recently added apps

### LINKED ISSUE #11086: app store: recently added
similar as most popular, lets have:

image

below most popular

### PR DESCRIPTION
**Pull request by Sahil25061999**
## What does this PR do?

Add recently added apps on app store page, it sorts the apps based on createdAt column.

Fixes #11086

image

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/app-store/_appRegistry.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ export async function getAppRegistry() {
     // Skip if app isn't installed
     /* This is now handled from the DB */
     // if (!app.installed) return apps;
-
+    app.createdAt = JSON.parse(JSON.stringify(dbapp.createdAt));
```
Comment (hariombalhara) [inline packages/app-store/_appRegistry.ts:47]:
```suggestion
    app.createdAt = app.createdA.toISOString()
```

File: packages/app-store/_appRegistry.ts [hunk 2]
```diff
@@ -101,6 +101,7 @@ export async function getAppRegistryWithCredentials(userId: number, userAdminTea
     // Skip if app isn't installed
     /* This is now handled from the DB */
     // if (!app.installed) return apps;
+    app.createdAt = JSON.parse(JSON.stringify(dbapp.createdAt));
```
Comment (hariombalhara) [inline packages/app-store/_appRegistry.ts:104]:
```suggestion
    app.createdAt = app.createdAt.toISOString()
```

It's more clear what's happening this way.

File: packages/types/App.d.ts [hunk 3]
```diff
@@ -153,6 +153,8 @@ export interface App {
   dependencies?: string[];
   /** Enables video apps to be used for team events. Non Video/Conferencing apps don't honor this as they support team installation always. */
   concurrentMeetings?: boolean;
+
```
Comment (hariombalhara) [inline packages/types/App.d.ts:156]:
```suggestion
```

**Review by hariombalhara — DISMISSED** [review]
Thanks @Sahil25061999 🙏 I pushed a fix and merging it now.

**Comment by hariombalhara** [comment]
There is still a TS error.`createdAt` is being considered required. You would have to make it optional. Could you handle that please.

**Comment by Sahil25061999** [comment]
yup will do it

**Comment by Sahil25061999** [comment]
@hariombalhara fixed the issue

File: packages/app-store/_appRegistry.ts [hunk 4]
```diff
@@ -44,7 +44,7 @@ export async function getAppRegistry() {
     // Skip if app isn't installed
     /* This is now handled from the DB */
     // if (!app.installed) return apps;
-
+    app.createdAt = dbapp.createdAt.toISOString();
```
Comment (PeerRich) [inline packages/app-store/_appRegistry.ts:47]: i always thought we already collect "createdAt" for apps, no? @hariombalhara

Comment (hariombalhara) [reply packages/app-store/_appRegistry.ts:47]: We were storing it, it's just that were not sending it to frontend.

**Review by hariombalhara — APPROVED** [review]

**Review by PeerRich — APPROVED** [review]

**Pull request merged**